### PR TITLE
Add prop-types package and update code

### PR DESCRIPTION
**Pull request by el-mapache**
* React.PropTypes is deprecated and was throwing a warning in the
console.

Just a lil day off coding..

There are a lot of changes but they can be boiled down to removing the `React` prefix from `PropTypes` and importing the `prop-types` library. I used [React codemod](https://github.com/reactjs/react-codemod) so I feel pretty confident about not missing anything.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,6 +114,7 @@
     "karma-jasmine-sinon": "^1.0.4",
     "lighthouse": "^1.6.3",
     "moxios": "^0.4.0",
+    "prop-types": "^15.5.10",
```
Comment (jcscottiii) [inline package.json]: would this be a general dependency and not just a dev dependency?

Comment (el-mapache) [inline package.json]: I don't think so, prop types are only checked locally, they don't have any bearing on production code.

**Comment by el-mapache** [comment]
Sorry, I'm not sure what you mean :( The package.json file is reporting the `prop-types` version is 15.5, do you mean I should update react to a newer version?

**Comment by msecret** [comment]
Oh I see what you mean now. That' actually kinda makes sense, because our package.json says use any version of react below v15. So we probably updated our deps and it updated shrinkwrap but doesn't have to update package.json.

**Comment by el-mapache** [comment]
@jcscottiii I think the '^' means that npm will try to update any minor or teeny versions of the package when it can, until the next major release. So, like @msecret said, when I regenerated the shrinkwrap file, npm saw there was a newer version of React (15.5.4) and picked that one over 15.4.2. Since it's a minor version there shouldn't be any breaking changes, and we should be safe to upgrade to that version of React when regenerating dependencies from the shrinkwrap file.

But, if we really want to have consistency, maybe we should peg the React version `package.json` and manually change it whenever we need to update React.

**Comment by jcscottiii** [comment]
That's true.

We can leave it as-is. Let's revisit this part when @msecret revisits the shrinkwrap part and if even matters since the shrinkwrap already pegs it.

**Pull request merged**